messageCheck/messageVerifier.go

```diff
@@ -106,29 +111,39 @@ func convertP2PMessagetoPubSubMessage(msg p2p.MessageP2P) (*pubsub_pb.Message, e
 	return newMsg, nil
 }
 
-func convertPubSubMessagestoP2PMessage(msg *pubsub_pb.Message) (p2p.MessageP2P, error) {
+func convertPubSubMessagestoP2PMessage(msg *pubsub_pb.Message, marshaller marshal.Marshalizer) (p2p.MessageP2P, error) {
 	if msg == nil {
 		return nil, p2p.ErrNilMessage
 	}
 
-	newMsg := &message.Message{
-		FromField:      msg.From,
-		PayloadField:   msg.Data,
-		SeqNoField:     msg.Seqno,
-		TopicField:     *msg.Topic,
-		SignatureField: msg.Signature,
-		KeyField:       msg.Key,
+	pubsubMsg := &pubsub.Message{
+		Message: &pubsub_pb.Message{
+			From:      msg.From,
+			Data:      msg.Data,
+			Seqno:     msg.Seqno,
+			Topic:     msg.Topic,
+			Signature: msg.Signature,
+			Key:       msg.Key,
+		},
+		ReceivedFrom:  "",
+		ValidatorData: nil,
+	}
+
+	newMsg, err := libp2p.NewMessage(pubsubMsg, marshaller)
```

Comment:
You can replace L132-L137 with: `return libp2p.NewMessage(pubsubMsg, marshaller)`

Reply:
done